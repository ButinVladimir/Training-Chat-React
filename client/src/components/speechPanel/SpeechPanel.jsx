import React from 'react';
import PropTypes from 'prop-types';
import NotLoginnedMessage from './notLoginnedMessage/NotLoginnedMessage';
import SpeechField from './speechField/SpeechFieldContainer';
import './style.css';

function SpeechPanel({
  loginned,
}) {
  return (
    <div className="chat-speech-panel-container">
      {!loginned && <NotLoginnedMessage />}
      {loginned && (
        <SpeechField />
      )}
    </div>
  );
}

SpeechPanel.propTypes = {
  loginned: PropTypes.bool.isRequired,
};

export default SpeechPanel;
