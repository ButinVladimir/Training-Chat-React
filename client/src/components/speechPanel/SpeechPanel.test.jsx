import React from 'react';
import { shallow } from 'enzyme';
import SpeechPanel from './SpeechPanel';

describe('SpeechPanel', () => {
  it('renders without crashing when not loginned', () => {
    const component = shallow(
      <SpeechPanel loginned={false} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when loginned', () => {
    const component = shallow(
      <SpeechPanel loginned />,
    );
    expect(component).toMatchSnapshot();
  });
});
