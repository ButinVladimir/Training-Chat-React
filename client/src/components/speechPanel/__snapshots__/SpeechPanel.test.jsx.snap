// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechPanel renders without crashing when loginned 1`] = `
<div
  className="chat-speech-panel-container"
>
  <Connect(SpeechField) />
</div>
`;

exports[`SpeechPanel renders without crashing when not loginned 1`] = `
<div
  className="chat-speech-panel-container"
>
  <NotLoginnedMessage />
</div>
`;
